pandas/core/indexes/base.py

```diff
@@ -3089,14 +3088,10 @@ def _deprecate_dti_setop(self, other: Index, setop: str_t) -> None:
             and other.tz is not None
```

Comment:
Could also skip the `tz_convert` if both `.tz`s are "UTC"

Reply:
we only get here with non-matching dtypes.  i'll follow up to make that clearer

Reply:
Ah gotcha. That'd be helpful, thanks